include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -0,0 +1,1280 @@
+ *
+ * \todo example
+ *
+ * \todo inserts still suffer from a range optimisation problem in GCC:
+ * https://github.com/ericniebler/range-v3/issues/700
+ *
+ * \par Exceptions
+ *
+ * Whenever a strong exception guarantee is given for this class, it presumes that
+ * `std::is_nothrow_move_constructible<typename inner_type::value_type>` otherwise only basic exception safety can
+ * be assumed.
+ *
+ * \par Thread safety
+ *
+ * This container provides no thread-safety beyond the promise given also by the STL that all
+ * calls to `const` member function are safe from multiple threads (as long as no thread calls
+ * a non-`const` member function at the same time).
+ *
+ */
+template <typename inner_type,
+          typename data_delimiters_type = std::vector<typename inner_type::size_type>>
+//!\cond
+    requires reservable_sequence_concept<std::remove_reference_t<inner_type>> &&
+             reservable_sequence_concept<std::remove_reference_t<data_delimiters_type>> &&
+             std::is_same_v<ranges::size_type_t<inner_type>, ranges::value_type_t<data_delimiters_type>>
+//!\endcond
+class concatenated_sequences
+{
+protected:
+    //!\privatesection
+    //!\brief Where the concatenation is stored.
+    std::decay_t<inner_type> data_values;
+    //!\brief Where the delimiters are stored; begins with 0, has size of size() + 1.
+    data_delimiters_type data_delimiters{0};
+
+public:
+    //!\publicsection
+    /*!\name Member types
+     * \{
+     */
+    //!\brief == inner_type.
+    //!\hideinitializer
+    using value_type = std::decay_t<inner_type>;
+
+    //!\brief A proxy of type ranges::view::slice that represents the range on the concatenated vector.
+    //!\hideinitializer
+    using reference = decltype(data_values | ranges::view::slice(0, 1));
+
+    //!\brief An immutable proxy of type ranges::view::slice that represents the range on the concatenated vector.
+    //!\hideinitializer
+    using const_reference = decltype(std::as_const(data_values) | ranges::view::slice(0, 1) | ranges::view::const_);
+
+    //!\brief The iterator type of this container (a random access iterator).
+    //!\hideinitializer
+    using iterator = detail::random_access_iterator<concatenated_sequences>;
+
+    //!\brief The const iterator type of this container (a random access iterator).
+    //!\hideinitializer
+    using const_iterator = detail::random_access_iterator<concatenated_sequences const>;
+
+    //!\brief A signed integer type (usually std::ptrdiff_t)
+    //!\hideinitializer
+    using difference_type = std::ptrdiff_t;
+
+    //!\brief An unsigned integer type (usually std::size_t)
+    //!\hideinitializer
+    using size_type = size_t;
+    //!\}
+
+    /* rule of six */
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Default constructors.
+    concatenated_sequences() = default;
+    //!\brief Default constructors.
+    constexpr concatenated_sequences(concatenated_sequences const &) = default;
```

Comment:
Ok, so there is no benefit 🤣 
I mean yes, it is potentially right. But the use case would be not really serving the class's functionality, since you can not fill the object at compile time with data.
But anyhow, I agree that it would do no harm to add `constexpr` specifier here.